psi4/driver/driver.py

```diff
@@ -873,6 +873,158 @@ def properties(*args, **kwargs):
         return core.variable('CURRENT ENERGY')
 
 
+
+def optimize_geometric(name, **kwargs):
+
+    import geometric
+    import qcelemental as qcel
+
+    class Psi4NativeEngine(geometric.engine.Engine):
+        """
+        Run a Psi4 energy and gradient calculation for geometric without any I/O
+        """
+        def __init__(self, p4_name, p4_mol, **p4_kwargs):
+    
+            self.p4_name = p4_name
+            self.p4_mol = p4_mol
+            self.p4_kwargs = p4_kwargs
+    
+            molecule = geometric.molecule.Molecule()
+            molecule.elem = [p4_mol.symbol(i) for i in range(p4_mol.natom())]
+            molecule.xyzs = [p4_mol.geometry().np * qcel.constants.bohr2angstroms] # reshape?
+            molecule.build_bonds()
+                                 
+            super(Psi4NativeEngine, self).__init__(molecule)
+    
+        def calc(self, coords, dirname):
+            self.p4_mol.set_geometry(core.Matrix.from_array(coords.reshape(-1,3)))
+            self.p4_mol.update_geometry()
+            g, wfn = gradient(self.p4_name, return_wfn=True, molecule=self.p4_mol, **self.p4_kwargs)
+            e = core.variable('CURRENT ENERGY')
+            return {'energy': e, 'gradient': (g.np).reshape(-1)}
```

Comment:
Wouldn't `(g.np).ravel()` be a bit clearer? Do we have a style policy on `reshape`s and `ravel`s?

Reply:
I wasn't aware of `.ravel()`, it's definitely clearer